src/main/java/it/gov/pagopa/payment/connector/rest/reward/mapper/AuthPaymentMapper.java

```diff
@@ -0,0 +1,58 @@
+package it.gov.pagopa.payment.connector.rest.reward.mapper;
+
+import it.gov.pagopa.payment.connector.rest.reward.dto.AuthPaymentRequestDTO;
+import it.gov.pagopa.payment.connector.rest.reward.dto.AuthPaymentResponseDTO;
+import it.gov.pagopa.payment.dto.AuthPaymentDTO;
+import it.gov.pagopa.payment.model.TransactionInProgress;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AuthPaymentMapper {
+
+  public AuthPaymentRequestDTO rewardMap(TransactionInProgress transactionInProgress) {
+    return AuthPaymentRequestDTO.builder()
+        .transactionId(transactionInProgress.getId())
+        .userId(transactionInProgress.getUserId())
+        .merchantId(transactionInProgress.getMerchantId())
+        .senderCode(transactionInProgress.getSenderCode())
+        .merchantFiscalCode(transactionInProgress.getMerchantFiscalCode())
+        .vat(transactionInProgress.getVat())
+        .idTrxAcquirer(transactionInProgress.getIdTrxAcquirer())
+        .trxDate(transactionInProgress.getTrxDate())
+        .amountCents(transactionInProgress.getAmountCents())
+        .amountCurrency(transactionInProgress.getAmountCurrency())
+        .mcc(transactionInProgress.getMcc())
+        .acquirerCode(transactionInProgress.getAcquirerCode())
+        .acquirerId(transactionInProgress.getAcquirerId())
+        .idTrxIssuer(transactionInProgress.getIdTrxIssuer())
+        .operationType(transactionInProgress.getOperationTypeTranscoded())
+        .trxChargeDate(transactionInProgress.getTrxChargeDate())
+        .correlationId(transactionInProgress.getCorrelationId())
+        .hpan(transactionInProgress.getHpan())
+        .build();
+  }
+
+  public AuthPaymentDTO rewardResponseMap(AuthPaymentResponseDTO responseDTO,
+      TransactionInProgress transactionInProgress) {
+    return AuthPaymentDTO.builder()
+        .id(responseDTO.getTransactionId())
+        .reward(responseDTO.getReward())
+        .initiativeId(responseDTO.getInitiativeId())
+        .rejectReasons(responseDTO.getRejectionReasons())
+        .status(responseDTO.getStatus())
+        .trxCode(transactionInProgress.getTrxCode())
+        .build();
+  }
+
+  public AuthPaymentDTO trxIdempotence(TransactionInProgress transaction){
```

Comment:
this is an uggly name :D let's use a name more general
```suggestion
  public AuthPaymentDTO transactionMapper(TransactionInProgress transaction){
```